File: serenity-app/src/test/java/us/nineworlds/serenity/ui/views/statusoverlayview/StatusOverlayFrameLayoutTest.java
package us.nineworlds.serenity.ui.views.statusoverlayview;

import android.app.Activity;
import org.assertj.core.api.Java6Assertions;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;
import us.nineworlds.serenity.BuildConfig;

import static org.assertj.android.api.Assertions.assertThat;

@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class)
public class StatusOverlayFrameLayoutTest {

  Activity activity;
  StatusOverlayFrameLayout view;

  @Before public void setUp() {
    activity = Robolectric.buildActivity(Activity.class).create().visible().get();
    view = new StatusOverlayFrameLayout(activity);
  }

  @After public void tearDown() {
    activity.finish();
    if (view.presenter != null) {
      view.onDetachedFromWindow();
    }
  }

  @Test public void verifyViewsAreInflated() {
    assertThat(view.infoGraphicMetaContainer).isNotNull();
    assertThat(view.metaOverlay).isNotNull();
    assertThat(view.posterInProgressIndicator).isNotNull();
    assertThat(view.posterOverlayTitle).isNotNull();
    assertThat(view.posterWatchedIndicator).isNotNull();
    assertThat(view.roundedImageView).isNotNull();
    assertThat(view.subtitleIndicator).isNotNull();
    assertThat(view.trailerIndicator).isNotNull();
  }

  @Test public void verifyPresenterSet() {
    Java6Assertions.assertThat(view.presenter).isNotNull();
  }
}

File: serenity-app/src/test/kotlin/us/nineworlds/serenity/ui/views/statusoverlayview/StatusOverlayFrameLayoutTest.kt
package us.nineworlds.serenity.ui.views.statusoverlayview

import android.app.Activity
import assertk.assertThat
import assertk.assertions.isNotNull
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class StatusOverlayFrameLayoutTest {

    private lateinit var activity: Activity
    private lateinit var view: StatusOverlayFrameLayout

    @Before
    fun setUp() {
        activity = Robolectric.buildActivity(Activity::class.java).create().visible().get()
        view = StatusOverlayFrameLayout(activity)
    }

    @After
    fun tearDown() {
        activity.finish()
        if (view.presenter != null) {
            view.onDetachedFromWindow()
        }
    }

    @Test
    fun verifyViewsAreInflated() {
        assertThat(view.infoGraphicMetaContainer).isNotNull()
        assertThat(view.metaOverlay).isNotNull()
        assertThat(view.posterInProgressIndicator).isNotNull()
        assertThat(view.posterOverlayTitle).isNotNull()
        assertThat(view.posterWatchedIndicator).isNotNull()
        assertThat(view.roundedImageView).isNotNull()
        assertThat(view.subtitleIndicator).isNotNull()
        assertThat(view.trailerIndicator).isNotNull()
    }

    @Test
    fun verifyPresenterSet() {
        assertThat(view.presenter).isNotNull()
    }
}
